src/DefinitionResolver.php

```diff
@@ -70,35 +86,77 @@ public function getDeclarationLineFromNode(Node $node): string
+        if ($constOrPropertyDeclaration !== null) {
+            $node = $constOrPropertyDeclaration;
+        }
+
+        // For parameters, parse the function-like declaration to get documentation for a parameter
+        if ($node instanceof Node\Parameter) {
+            $variableName = $node->getName();
+
+            $functionLikeDeclaration = ParserHelpers::getFunctionLikeDeclarationFromParameter($node);
+            $docBlock = $this->getDocBlock($functionLikeDeclaration);
+
+            $parameterDocBlockTag = $this->tryGetDocBlockTagForParameter($docBlock, $variableName);
+            return $parameterDocBlockTag !== null ? $parameterDocBlockTag->getDescription()->render() : null;
+        }
+
+        // For everything else, get the doc block summary corresponding to the current node.
+        $docBlock = $this->getDocBlock($node);
+        if ($docBlock !== null) {
+            // check wether we have a description, when true, add a new paragraph
+            // with the description
+            $description = $docBlock->getDescription()->render();
+
+            if (empty($description)) {
+                return $docBlock->getSummary();
             }
-        } else {
-            $docBlock = $node->getAttribute('docBlock');
-            if ($docBlock !== null) {
-                // check wether we have a description, when true, add a new paragraph
-                // with the description
-                $description = $docBlock->getDescription()->render();
-
-                if (empty($description)) {
-                    return $docBlock->getSummary();
-                }
-                return $docBlock->getSummary() . "\n\n" . $description;
+
+            return $docBlock->getSummary() . "\n\n" . $description;
+        }
+        return null;
+    }
+
+    /**
+     * Gets Doc Block with resolved names for a Node
+     *
+     * @param Node $node
+     * @return DocBlock | null
+     */
+    private function getDocBlock(Node $node)
+    {
+        // TODO make more efficient (caching, ensure import table is in right format to begin with)
+        $docCommentText = $node->getDocCommentText();
+        if ($docCommentText !== null) {
+            list($namespaceImportTable,,) = $node->getImportTablesForCurrentScope();
+            foreach ($namespaceImportTable as $alias => $name) {
+                $namespaceImportTable[$alias] = (string)$name;
+            }
+            $namespaceDefinition = $node->getNamespaceDefinition();
+            if ($namespaceDefinition !== null && $namespaceDefinition->name !== null) {
+                $namespaceName = (string)$namespaceDefinition->name->getNamespacedName();
+            } else {
+                $namespaceName = 'global';
+            }
+            $context = new Types\Context($namespaceName, $namespaceImportTable);
+
+            try {
+                return $this->docBlockFactory->create($docCommentText, $context);
+            } catch (\InvalidArgumentException $e) {
```

Comment:
I see it thrown a few times if I remove the try/catch. One example: a doccomment line in the cakephp project source is:

`* @see config/app.php for configuration settings`

We pass all the docCommentText, and the factory throws because config/app.php is not a valid fqsen. That seems like the wrong thing for phpdocumentor to do? I don't think we can validate it before passing it in. I'll add a comment though.